### ISSUE TITLE
Prism launcher allocating 4 GB's for PrismLauncher-0.log

### ISSUE THREAD

**Issue by iblowmymind** [PrismLauncher/PrismLauncher#752]
### Operating System

Windows

### Version of Prism Launcher

Prism Launcher 6.1

### Version of Qt

Qt 6.4.0

### Description of bug

When an instance is launched, Prism Launcher will make the size on disk of file PrismLauncher-0.log 4GB's, effectively reducing the free disk space by 4 GB's. When the instance is closed, this space is not reclaimed, but when Prism's exited as well, the file will return back to normal (~40 KB's), immediately releasing 4 GB's from used disk space.
![CopyQ nityPu](https://user-images.githubusercontent.com/28228415/212532777-8e55fbb3-557b-4a17-aa40-eff6d514e877.png)
When the file is opened with Notepad++, its contents are normal and as expected (e.g., there are no invisible NUL characters filling up the file)

Observing the file with WizTree, while Prism's not running and after launching it (not running any instance yet):
![CopyQ OGdYIn](https://user-images.githubusercontent.com/28228415/212532958-90bdcb98-a784-43cc-b21d-d57ca2fe763d.png)
Immediately after launching any instance (and after closing it but not closing Prism):
![CopyQ DMVcxK](https://user-images.githubusercontent.com/28228415/212532916-0024db2d-80ec-470c-a6b6-1e917130bb23.png)
And, finally closing Prism as well:
![CopyQ rCFhUV](https://user-images.githubusercontent.com/28228415/212533037-57a3f858-a136-45f9-9154-ab6b6ad31421.png)

Note: Explorer will still report the size to be 30-40 KB's no matter what, but right-clicking and viewing size on disk will reveal that it's 4 GB's. I initially suspected this to be related to my pagefile growing when I'm low on memory while I try to launch Minecraft, but the size of my pagefile doesn't change.

### Steps to reproduce

* Launch Prism Launcher
* Launch an instance (in my case, a Quilt instance running 1.19.3 with some mods)
* Watch as the disk space in your drive goes down by 4 GB's
* Close Minecraft (disk space not reclaimed)
* Finally, close Prism (disk space reclaimed)

### This issue is unique

- [x] I have searched the issue tracker and did not find an issue describing my bug.

**Comment by iblowmymind** [comment]
Still getting this bug with 6.3. Opening the log file and saving it without any changes fixes the issue while Prism is running as a temporary workaround.

**Comment by Jorma17** [comment]
I have a similar issue, but the log file gets **actually** filled. In my PrismLauncher directory I have a file named `PrismLauncher-0.log` that is 6.8 GiB or 7.2 GB in size. The file has almost 148 million lines of text. At the start there's some normal text like "Log initialized" and "Loading account" but after line 85 every line is just some number a.bcd followed by a capital D and then the text "Checking  "./PrismLauncher-0.log". As I previously said, this spans for around 148 million lines.

I'll paste a small portion of it here:
```
0.119 D | Checking  "./PrismLauncher-0.log"
0.119 D | Checking  "./PrismLauncher-0.log"
0.119 D | Checking  "./PrismLauncher-0.log"
0.119 D | Checking  "./PrismLauncher-0.log"
0.119 D | Checking  "./PrismLauncher-0.log"
0.119 D | Checking  "./PrismLauncher-0.log"
0.119 D | Checking  "./PrismLauncher-0.log"
0.119 D | Checking  "./PrismLauncher-0.log"
0.119 D | Checking  "./PrismLauncher-0.log"
0.119 D | Checking  "./PrismLauncher-0.log"
0.119 D | Checking  "./PrismLauncher-0.log"
0.119 D | Checking  "./PrismLauncher-0.log"
0.119 D | Checking  "./PrismLauncher-0.log"
0.120 D | Checking  "./PrismLauncher-0.log"
0.120 D | Checking  "./PrismLauncher-0.log"
0.120 D | Checking  "./PrismLauncher-0.log"
0.120 D | Checking  "./PrismLauncher-0.log"
0.120 D | Checking  "./PrismLauncher-0.log"
0.120 D | Checking  "./PrismLauncher-0.log"
0.120 D | Checking  "./PrismLauncher-0.log"
0.120 D | Checking  "./PrismLauncher-0.log"
0.120 D | Checking  "./PrismLauncher-0.log"
0.120 D | Checking  "./PrismLauncher-0.log"
0.120 D | Checking  "./PrismLauncher-0.log"
0.120 D | Checking  "./PrismLauncher-0.log"
0.120 D | Checking  "./PrismLauncher-0.log"
0.120 D | Checking  "./PrismLauncher-0.log"
0.120 D | Checking  "./PrismLauncher-0.log"
0.120 D | Checking  "./PrismLauncher-0.log"
```

This goes on and on, with the number in front increasing all the time. I am using Arch Linux and the flatpak version of Prism Launcher after the AUR one stopped working when they updated QT. I have no idea if this happened before or after my switch.

**Comment by Scrumplex** [comment]
@Jorma17 judging by the message, I think it is coming from here: https://github.com/PrismLauncher/PrismLauncher/blob/8dc3267925537d58a1e9418863b78e4cc55bd1c1/launcher/icons/IconList.cpp#L187

Is is possible that your icons folder is set to an empty value in the settings?

![icons setting](https://scrumplex.rocks/img/1682102081_Fal9oo.png)

**Comment by Jorma17** [comment]
No, I checked and I had it at `/home/[username]/.local/share/PrismLauncher/` and that might be the cause of the issue. I changed it to `/home/[username]/.local/share/PrismLauncher/icons/`.  I also moved the log file away and will now check that file regularly to see if anything happens.

EDIT: The file `PrismLauncher-0.log` has not reappeared but something really strange happened. The original `PrismLauncher-0.log`, which I moved to another location, is now only 18 MiB in size. However, now the file `PrismLauncher-1.log` is 6.8 GiB instead of the original 18 MiB and the same issue occurs there. It's mostly just `Checking "./PrismLauncher-0.log"` with the increasing number in front. This is quite weird.

**Comment by iblowmymind** [comment]
> Is is possible that your icons folder is set to an empty value in the settings?

In my end, it wasn't- it's set to the default of "icons".

**Comment by Scrumplex** [comment]
> EDIT: The file `PrismLauncher-0.log` has not reappeared but something really strange happened. The original `PrismLauncher-0.log`, which I moved to another location, is now only 18 MiB in size. However, now the file `PrismLauncher-1.log` is 6.8 GiB instead of the original 18 MiB and the same issue occurs there. It's mostly just `Checking "./PrismLauncher-0.log"` with the increasing number in front. This is quite weird.

Expected behavior.

Every time you start the launcher, it rotates the log files. So PrismLauncher-4.log becomes -5.log, -3 becomes -4.log and so on.

The reason why it spams `Checking "./PrismLauncher-0.log"` is that it is constantly being updated. Every time it prints `Checking ...`, the file gets updated, which causes it to check the file again.

This shouldn't happen once you fix that icons path, though.

**Comment by Scrumplex** [comment]
> > Is is possible that your icons folder is set to an empty value in the settings?
> 
> In my end, it wasn't- it's set to the default of "icons".

Can you try to check the contents of the file, before the space is reclaimed?

**Comment by iblowmymind** [comment]
The contents of the file are as usual. (No weird characters, no nothing) I don't believe it's something inside the file, anyway; as explorer reports the file size as ~40-100KBs but 4 GBs on disk. Opening the document and saving it without modifications makes it 40KBs on disk as well. Is the launcher accidentally creating a sparse file somehow?

**Comment by RaptaG** [comment]
So the problem is an invalid icon path?

**Comment by iblowmymind** [comment]
I believe that issue's unrelated to mine, but the issue on my end seems to have resolved with 7.1/7.2. I'll test further and see if my issue persists, and if not, will close this issue. I think a seperate issue should be created for @Jorma17 's problem, though.
